WEB-INTEGRATION.md

```diff
@@ -397,53 +388,62 @@ registration context; that is, the context in which the class is constructed.
+`PerformanceEntry` would expose the snapshot as a `resourceContext` property.
 
 ## Events
 
 Events are a single API that is used for a great number of things, including
-cases which have a causal context (for events, also referred to as the dispatch
-context) separate from the registration context, and cases which have no
-dispatch context at all.
+cases which have a causal context (for events, also referred to as the
+**dispatch context**) separate from the registration context, and cases which
+have no dispatch context at all.
 
 For consistency, event listener callbacks should be called with the dispatch
 context. If that does not exist, the empty context should be used, where all
 `AsyncContext.Variable`s are set to their initial values.
 
-This use of the empty context, however, clashes with the goal of allowing
-“isolated” regions of code that share an event loop, and being able to trace
-in which region an error originates. A solution to this would be the ability to
-define a fallback context for a region of code. We have a proposal for this
-being fleshed out at issue
+Event dispatches can be one of the following:
+- **Synchronous dispatches**, where the event dispatch happens synchronously
+  when a web API is called. Examples are `el.click()` which synchronously fires
+  a `click` event, setting `location.hash` which synchronously fires a
+  `popstate` event, or calling an `EventTarget`'s `dispatchEvent()` method. For
+  these dispatches, the TC39 proposal's machinery is enough to track the
+  dispatch context, with no help from web specs or browser engines.
+- **Browser-originated dispatches**, where the event is triggered by browser or
+  user actions, or by cross-agent JS, with no involvement from JS code in the
+  same agent. Such dispatches can't have any dispatch context, so the listener
+  is called with the empty context. (Though see the section on fallback context
+  below.)
+- **Asynchronous dispatches**, where the event originates from JS calling into
+  some web API, but the dispatch happens at a later point. In these cases, the
+  context should be tracked along the data flow of the operation, even across
+  code running in parallel (but not through tasks enqueued on other agents'
+  event loops). [See below on implicit context
+  propagation](#implicit-context-propagation) for how this data flow tracking
+  should happen.
+
+This classification of event dispatches is the way it should be in theory, as
+well as a long-term goal. However, as we describe later in the section on
+implicit context propagation, for the initial rollout we propose treating the
+vast majority of asynchronous dispatches as if they were browser-originated.
+The exceptions would be:
+
+- The `popstate` event
+- The `message` and `messageerror` events
+- All events dispatched on `XMLHttpRequest` or `XMLHttpRequestUpload` objects
+- The `error`, `unhandledrejection` and `rejectionhandled` events on the global
+  object (see below)
```

Comment:
@shaseley @mmocny Are there any other events that task attribution is explicitly propagating?